packages/gatsby-source-prismic-graphql/src/gatsby-node.ts

```diff
@@ -224,12 +227,16 @@ exports.createPages = async ({ graphql, actions: { createPage } }: any, options:
     documents: Edge[] = []
   ): Promise<Edge[]> {
     // Format page.type so that the graphql query doesn't complain.
-    const pageTypeUnderscored = page.type.toLowerCase().split(' ').join('_');
-    const pageTypeFormatted = pageTypeUnderscored.charAt(0).toUpperCase() + pageTypeUnderscored.slice(1);
+    const pageTypeUnderscored = page.type
+      .toLowerCase()
+      .split(' ')
+      .join('_');
+    const pageTypeFormatted =
+      pageTypeUnderscored.charAt(0).toUpperCase() + pageTypeUnderscored.slice(1);
     // Prepare and execute query
     const documentType: string = `all${pageTypeFormatted}s`;
     const sortType: string = `PRISMIC_Sort${pageTypeFormatted}y`;
-    const extraPageFields = options.extraPageFields || '';
+    const extraPageFields = page.extraPageFields || options.extraPageFields || '';
```

Comment:
frankly I think `options.extraPageFields` should be removed (it is "extra _page_ fields" after all), but didn't want to introduce a breaking change.

Reply:
hmm, good point